great_expectations/datasource/fluent/snowflake_datasource.py

```diff
@@ -218,10 +229,14 @@ def get_engine(self) -> sqlalchemy.Engine:
                 connection_string: str | dict = model_dict.pop("connection_string")
 
                 if isinstance(connection_string, str):
-                    self._engine = sa.create_engine(connection_string, **kwargs)
+                    self._engine = sa.create_engine(
+                        connection_string,
+                        connect_args={"application": snowflake_partner_application},
```

Comment:
Just eyeballing this: does this conflict if the user has `connect_args` in their kwargs?

Reply:
Ah great catch! Yeah, would probably break. Will update to merge if present.

Reply:
Or actually, just noticed a test that suggests providing both str and connect args should yield ValueError. Lemme dig deeper and confirm expected behavior.

Reply:
Changed impl to add query param to url instead of using connect_args, mostly to allow for easier testing (once an engine is instantiated, there's no way to really check connect_args without actually connecting to Snowflake).